docs/20_smart-contracts/30_tutorials/10_creating-a-token.md

```diff
@@ -0,0 +1,631 @@
+and the `asset.amount` property (which is of type `int64_t`).
+
+The `primary_key` function in the structure is used to uniquely identify each row for indexing purposes. In this case, 
+we are using the `owner` field as the primary key, but using the `uint64_t` representation instead for efficiency.
+
+Next, you need another table to store the total supply of the token. Add this below the `balances_table` you just added:
+
+```c++
+    using supply_table = singleton<"supply"_n, asset>;
+```
+
+We're using a different type of table here, a `singleton`. A `singleton` is a table that only has one row per-scope. 
+This is perfect for storing things like configurations. We will use it to store the total supply of the token as you 
+only have the one token in your contract.
+
+You can see that we also didn't define a structure to store as we only need the `asset` type to store the total supply,
+though you _could_ define a structure if you wanted to.
+
+## Filling in the actions
+
+Now that you have your data structures defined, let's fill in the actions.
+
+### Issue
+
+First we'll start with the `issue` action, which will create new tokens and add them to an account's balance.
+
+We want only the account that the contract is deployed on to be able to call the `issue` action, so we will add 
+an assertion to make sure that the account calling the action is the same as the account that the contract is deployed on.
+
+```c++
+    ACTION issue(name to, asset quantity){
+        check(has_auth(get_self()), "only contract owner can issue new GOLD");
+    }
+```
+
+Next, we want to make sure that the account we will issue the tokens to exists on the blockchain. We don't want that
+sweet in-game GOLD to go to waste!
+
+```c++
+    ...
+    check(is_account(to), "the account you are trying to issue GOLD to does not exist");
+```
+
+Next, we want to make sure that the `quantity` parameter is a positive number, and has the correct 
+`symbol` and `precision`.
+
+```c++
+    ...
+    check(quantity.is_valid(), "invalid quantity");
+    check(quantity.amount > 0, "must issue a positive quantity");
+    check(quantity.symbol == TOKEN_SYMBOL, "symbol precision and/or ticker mismatch");
+```
+
+Shwew! That's a lot of checks, but it's important to make sure that we are protecting our in-game GOLD!
+
+Now let's start dealing with the balances table.
+
+```c++
+    ...
+    balances_table balances(get_self(), get_self().value);
+```
+
+We took the `balances_table` type we defined earlier and instantiated a new `balances_table` object. We passed in the 
+`get_self()` function as the first parameter, which returns the contract account's name. We passed in the `get_self().value`
+as the second parameter, which returns the `uint64_t` representation of the contract account's name. This is the scope 
+that the `balances` table will be stored in (scope is just a way to partition data within a table). 
```

Comment:
I find this confusing. The first part of the sentence is saying the table is placed in a specific scope. And the second part is saying the table is partitioned.
it is not easy to talk about code and scope in two short sentences, I know. but it is a quite important concept the dev has to grasp well when working with tables.

idea: at the end of the doc we could have a section "see also" and have a link to https://docs.eosnetwork.com/docs/latest/smart-contracts/getting-started/smart_contract_basics#multi-index-instantiate-with-code-and-scope